configure.ac

```diff
@@ -1036,4 +1040,5 @@ Now type 'make @<:@<target>@:>@'
     check        - run the tests
     doxygen-doc  - generate the HTML documentation
     lint         - run the linters
+    coverage     - generate coverage post --enable-gcov
```

Comment:
I think this should be able to force --enable-gcov like we do with say ja-rule and the USB stuff.

Reply:
`--enable-gcov` is a `configure` flag, but running `make coverage` would happen after the fact. Are you saying that running `make coverage` should re-run `configure` with `--enable-gcov` added?

Reply:
Ah gotcha. Yeah maybe not that, but presumably we could at least throw a different error if `--enable-gcov` wasn't passed to configure and we call `make coverage`, given it presumably won't work in this case.

Reply:
Ah yes, we can certainly catch that. I am honestly not very familiar with automake so I'm not certain how to do that.

Reply:
OK, I think I got this to work. Running `make coverage` should now build if it needs to and then run the coverage generation (similar to how `make check` builds and runs unit tests).

Reply:
OK, this is passing! Running `make check` runs the checks and then the coverage. I couldn't figure out how to get it to compile the checks without running them, but realistically this isn't very necessary.